libvast/src/system/partition_transformer.cpp

```diff
@@ -291,41 +345,53 @@ partition_transformer_actor::behavior_type partition_transformer(
     },
     [self](atom::internal, atom::resume, atom::done, vast::id offset) {
       VAST_DEBUG("partition-transformer got new offset {}", offset);
-      self->state.data.offset = offset;
-      for (auto& slice : self->state.slices) {
-        slice.offset(offset);
-        offset += slice.rows();
-        self->state.add_slice(slice);
-        self->state.stage->out().push(slice);
+      for (auto& [_, data] : self->state.data) {
+        data.offset = offset;
+        for (auto& slice : self->state.slices) {
+          slice.offset(offset);
+          offset += slice.rows();
+          self->state.add_slice(slice);
+          self->state.stage->out().push(slice);
+        }
       }
       self->state.finalize_data();
       detail::shutdown_stream_stage(self->state.stage);
       auto stream_data = partition_transformer_state::stream_data{};
       [&] {
-        { // Pack partition
+        stream_data.partition_chunks
+          = std::vector<std::tuple<vast::uuid, vast::type, vast::chunk_ptr>>{};
```

Comment:
The vector should already be default initialized from line 359.

Reply:
Actually it's not, the field is a `caf::expected<std::vector<...>>`, so the default is to initialize it as an error.